### DISCUSSION TITLE
How the new methods would change the implementation of graphql/apollo?

### DISCUSSION THREAD

**Discussion by samuelcastro** [vercel/next.js#12182] [Help]
So with those new methods we have available  (`getStaticProps`,  `getStaticPaths`, `getServerSideProps`) is it going to change somehow the way to handle data fetch using graphql/apollo?

Thanks!

**Comment by benjaminpearson** [comment]
I know this question is old and hopefully you found an answer, just thought I would share this example https://github.com/vercel/next.js/pull/13202 if you are still looking or for anyone else who lands on this discussion in future - it shows a way to query a local GraphQL schema in `getStaticProps` & `getStaticPaths`. 

For remote GraphQL endpoints there are several examples in the examples folder, particularly ones related to CMS SaaS products such as Dato https://github.com/vercel/next.js/tree/canary/examples/cms-datocms

**Reply by samuelcastro** [reply]
Thanks @benjaminpearson.
